src/algorithms/fardetectors/MatrixTransferStatic.cc

```diff
@@ -28,41 +28,153 @@ void eicrecon::MatrixTransferStatic::init(const dd4hep::Detector* det,
   m_detector  = det;
   m_converter = id_conv;
   //Calculate inverse of static transfer matrix
+    
+}
+
+void eicrecon::MatrixTransferStatic::process(
+    const MatrixTransferStatic::Input& input,
+    const MatrixTransferStatic::Output& output) const {
+
+  const auto [mcparts, rechits] = input;
+  auto [outputParticles] = output;
+  
   std::vector<std::vector<double>> aX(m_cfg.aX);
   std::vector<std::vector<double>> aY(m_cfg.aY);
+  
+  //----- Define constants here ------
+  double aXinv[2][2] = {{0.0, 0.0},
+                        {0.0, 0.0}};
+  double aYinv[2][2] = {{0.0, 0.0},
+                        {0.0, 0.0}};
+  
+  double nomMomentum     = m_cfg.nomMomentum; //extract the nominal value first -- will be overwritten by MCParticle
+  double local_x_offset  = m_cfg.local_x_offset;
+  double local_y_offset  = m_cfg.local_y_offset;
+  double local_x_slope_offset  = m_cfg.local_x_slope_offset;
+  double local_y_slope_offset  = m_cfg.local_y_slope_offset;
+ 
+  double numBeamProtons = 0;
+  double runningMomentum = 0.0;
+  
+  for (const auto& p: *mcparts) {
+  	if (p.getGeneratorStatus() == 4 && p.getPDG() == 2212) { //look for "beam" proton
+		runningMomentum += p.getMomentum().z;
+		numBeamProtons++;
+	}
+  }
+  
+  if(numBeamProtons == 0) {m_log->error("No beam protons to choose matrix!! Skipping!!"); return;}
```

Comment:
From recent presentations/discussions it seems people are still wanting to run particle gun simulations, or run other particle species which could be corrected for afterwards? It might be worth adding a command line config parameter for this to fall back on if there's no beam protons.

Reply:
Also if #1251 gets merged we can pre-create a collection to pass only containing the beam protons.

Reply:
Hey Simon,

That's a good point. I am adding an additional check for the case where
MCParticles.size() == 1, which would indicate a particle gun study. I will
push the commit in a few minutes.

Reply:
I should note, this is only for protons right now. The only other case(s) to deal with are deuterons and He4, which I need to think of a solution for in terms of the correct matrices + beams. Since I do not have a solution for that on-hand, that will be for a different PR.